js/modules/k6/experimental/streams/module.go

```diff
@@ -0,0 +1,285 @@
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{Named: map[string]interface{}{
+		"ReadableStream":              mi.NewReadableStream,
+		"CountQueuingStrategy":        mi.NewCountQueuingStrategy,
+		"ReadableStreamDefaultReader": mi.NewReadableStreamDefaultReader,
+	}}
+}
+
+// NewReadableStream is the constructor for the ReadableStream object.
+func (mi *ModuleInstance) NewReadableStream(call goja.ConstructorCall) *goja.Object {
+	rt := mi.vu.Runtime()
+	var err error
+
+	// 1. If underlyingSource is missing, set it to null.
+	var underlyingSource *goja.Object
+
+	var (
+		strategy             *goja.Object
+		underlyingSourceDict UnderlyingSource
+	)
+
+	// We look for the queuing strategy first, and validate it before
+	// the underlying source, in order to pass the Web Platform Tests
+	// constructor tests.
+	strategy = mi.initializeStrategy(call)
+
+	// 2. Let underlyingSourceDict be underlyingSource, converted to an IDL value of type UnderlyingSource.
+	if len(call.Arguments) > 0 && !goja.IsUndefined(call.Arguments[0]) {
+		// We first assert that it is an object (requirement)
+		if !isObject(call.Arguments[0]) {
+			throw(rt, newError(TypeError, "underlyingSource must be an object"))
+		}
+
+		// Then we try to convert it to an UnderlyingSource
+		underlyingSource = call.Arguments[0].ToObject(rt)
+		underlyingSourceDict, err = NewUnderlyingSourceFromObject(rt, underlyingSource)
+		if err != nil {
+			throw(rt, err)
+		}
+	}
+
+	// 3. Perform ! InitializeReadableStream(this).
+	stream := &ReadableStream{
+		runtime: mi.vu.Runtime(),
+		vu:      mi.vu,
+	}
+	stream.initialize()
+
+	// 4. If underlyingSourceDict["type"] is "bytes":
+	if underlyingSourceDict.Type == "bytes" {
+		common.Throw(stream.runtime, newError(RuntimeError, "'bytes' stream is not supported yet"))
+	} else { // 5. Otherwise,
+		// 5.1. Assert: underlyingSourceDict["type"] does not exist.
+		if underlyingSourceDict.Type != "" {
+			common.Throw(rt, newError(AssertionError, "type must not be set for non-byte streams"))
+		}
+
+		// 5.2. Let sizeAlgorithm be ! ExtractSizeAlgorithm(strategy).
+		sizeAlgorithm := extractSizeAlgorithm(rt, strategy)
+
+		// 5.3. Let highWaterMark be ? ExtractHighWaterMark(strategy, 1).
+		highWaterMark := extractHighWaterMark(rt, strategy, 1)
+
+		// 5.4. Perform ? SetUpReadableStreamDefaultControllerFromUnderlyingSource(...).
+		stream.setupReadableStreamDefaultControllerFromUnderlyingSource(
+			underlyingSource,
+			underlyingSourceDict,
+			highWaterMark,
+			sizeAlgorithm,
+		)
+	}
+
+	streamObj := rt.ToValue(stream).ToObject(rt)
+
+	proto := call.This.Prototype()
+	if proto.Get("locked") == nil {
+		err = proto.DefineAccessorProperty("locked", rt.ToValue(func() goja.Value {
+			return rt.ToValue(stream.Locked)
+		}), nil, goja.FLAG_FALSE, goja.FLAG_TRUE)
+		if err != nil {
+			common.Throw(rt, newError(RuntimeError, err.Error()))
+		}
+	}
```

Comment:
I guess you have done this to past some tests using `instanceof` or comparing prototypes? 
But at least for me dropping it still passes the tests :thinking: 

Also in general this is done by creating a prototype object and then using that as goja does through out. Looking at the specification for the streams I can't find where this is defined as well as actually anything on prototypes. So I guess it might be in some of the connected specifications :grimacing: